src/AudioGeneratorMP3.cpp

```diff
@@ -118,13 +123,18 @@ enum mad_flow AudioGeneratorMP3::Input()
   }
 
   lastReadPos = file->getPos() - unused;
-  int len = buffLen - unused;
-  len = file->read(buff + unused, len);
-  if (len == 0) {
-    Serial.printf_P(PSTR("MP3 stop, len==0\n"));
-    return MAD_FLOW_STOP;
+  int readLen = buffLen - unused;
+  int len = file->read(buff + unused, readLen);
+  if (readLen != len) {
```

Comment:
Why isn't this on readLen==0 branch?  When streaming, the TCP layer may occasionally send a short read, need to think about what this would do as written...

Reply:
So, the reason I did it originally was to avoid any delay in trying to do another call to Input(). That may be unnecessary (it was part of many things I've tried). I'll see if, with my latest setup, I can roll this back to something more like it was.